### ISSUE TITLE
OS X: bold highlighting failing on autocompletion

### ISSUE THREAD

**Issue by robmcmullen** [robmcmullen/peppy#620]
_[Trac time 20090427 195926Z]_ When using the minibuffer for autocompletion, the bold text seems to be failing:

```
Traceback (most recent call last):
 File %22/Users/rob/src/peppy/peppy/lib/textctrl_autocomplete.py%22, line 387, in OnText
   self.processText(text)
 File %22/Users/rob/src/peppy/peppy/lib/textctrl_autocomplete.py%22, line 421, in processText
   self.dropdownlistbox.setBold(numCh, index, nchar)
 File %22/Users/rob/src/peppy/peppy/lib/textctrl_autocomplete.py%22, line 107, in setBold
   self._calcWidth(index)
 File %22/Users/rob/src/peppy/peppy/lib/textctrl_autocomplete.py%22, line 145, in _calcWidth
   w2, h2 = self.bold_dc.GetTextExtent(bold)
 File %22//usr/local/lib/wxPython-unicode-2.8.9.2/lib/python2.5/site-packages/wx-2.8-mac-unicode/wx/_gdi.py%22, line 3730, in GetTextExtent
   return _gdi_.DC_GetTextExtent(*args, **kwargs)
PyAssertionError: C++ assertion %22Ok()%22 failed at /BUILD/wxPython-src-2.8.9.2/src/common/dcgraph.cpp(872) in DoGetTextExtent(): wxGCDC(cg)::DoGetTextExtent - invalid DC
```

This can be demoed by trying to switch buffers using C-x b and attempting to switch to the "scratch" buffer.

**Comment by robmcmullen** [comment]
_[Trac time 20090427 205132Z]_ b20546b934b331f17713bcb1a80ac65f14bc3c4d Refs #620: added "ignore reoccurrences" option in error dialog to allow user to continue working

**Comment by robmcmullen** [comment]
_[Trac time 20090427 211646Z]_ 7eb2d17ddd44f138a2c0af0bc5b3e33742798d01 Fixed #620: OS X can't seem to use a MemoryDC to get text extents
- changed to use the wxWindow's local GetTextExtents by changing to either the bold or regular text before the call to GetTextExtents
